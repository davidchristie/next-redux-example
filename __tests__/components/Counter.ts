import Counter from "../../components/Counter";
import { decrement, increment, reset } from "../../actions";
import {
  describeComponent,
  describeWhenButtonIsClicked,
  itShouldDispatchAction,
  itShowsText,
  itShouldMatchSnapshot
} from "../../test-utils";

describeComponent(Counter, {}, getContext => {
  itShowsText(getContext, "Count:");

  itShowsText(getContext, "0");

  itShouldMatchSnapshot(getContext);

  describeWhenButtonIsClicked(getContext, "+1", getContext => {
    itShouldDispatchAction(getContext, increment());
  });

  describeWhenButtonIsClicked(getContext, "-1", getContext => {
    itShouldDispatchAction(getContext, decrement());
  });

  describeWhenButtonIsClicked(getContext, "Reset", getContext => {
    itShouldDispatchAction(getContext, reset());
  });
});
