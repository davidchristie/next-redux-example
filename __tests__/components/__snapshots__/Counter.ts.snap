// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-1118913787"
  >
    <h1
      class="jsx-1118913787"
    >
      Count: 
      <span
        class="jsx-1118913787"
      >
        0
      </span>
    </h1>
    <button
      class="jsx-1118913787"
    >
      +1
    </button>
    <button
      class="jsx-1118913787"
    >
      -1
    </button>
    <button
      class="jsx-1118913787"
    >
      Reset
    </button>
  </div>
</DocumentFragment>
`;
